packages/common/src/util/http.js

```diff
@@ -0,0 +1,135 @@
+import { Client, MockAgent } from 'undici';
+
+const clients = new Map();
+
+const getClient = (baseUrl, options) => {
+  const { tls, timeout } = options;
+  if (!clients.has(baseUrl)) {
+    clients.set(
+      baseUrl,
+      new Client(baseUrl, { bodyTimeout: timeout, connect: tls })
+    );
+  }
+  return clients.get(baseUrl);
+};
+
+export const enableMockClient = baseUrl => {
+  const mockAgent = new MockAgent({ connections: 1 });
+  const client = mockAgent.get(baseUrl);
+  if (!clients.has(baseUrl)) {
+    clients.set(baseUrl, client);
+  }
+  return client;
+};
+
+const assertOK = (response, errorMap, fullUrl) => {
+  const errMapMessage = errorMap[response.statusCode];
+
+  if (errMapMessage || response.statusCode >= 400) {
+    const defaultErrorMesssage = `Request to ${fullUrl} failed with status: ${response.statusCode}`;
+
+    const errMessage =
+      typeof errMapMessage === 'function'
+        ? errMapMessage(response)
+        : errMapMessage || defaultErrorMesssage;
+
+    const error = new Error(errMessage);
+    error.code = response.statusCode;
+    error.url = fullUrl;
+    throw error;
+  }
+};
+
+const parseUrl = (fullUrl, baseUrl) => {
+  if (/^https?:\/\//.test(fullUrl)) {
+    const url = new URL(fullUrl);
+    return {
+      baseUrl: url.origin,
+      path: url.pathname,
+    };
+  } else {
+    return { baseUrl, path: fullUrl };
+  }
+};
+
+const defaultOptions = {
+  timeout: 300e3, // Default to 300 seconds
+  headers: {},
+  query: undefined,
+  body: undefined,
+  errors: {},
+  tls: {},
+  parseAs: 'auto',
+};
+
+/**
```

Comment:
Request is an internal function. We should be describing (and more importantly the types of its arguments) it in typescript, probably in index.ts.

This function will not currently be generated in docs.openfn.org. And it probably shouldn't be because it's not used by job writers (only by adpator writers). So we don't actually need a full JSDoc here.